Objects/genericaliasobject.c

```diff
@@ -626,6 +640,65 @@ static PyNumberMethods ga_as_number = {
         .nb_or = _Py_union_type_or, // Add __or__ function
 };
 
+static PyObject *
+ga_iternext(gaiterobject *gi) {
+    gaobject *alias = (gaobject *) gi->obj;
+
+    if (gi->exhausted) {
+        gi->obj = NULL;
+        return NULL;
+    }
+    gi->exhausted = 1;
+
+    gaobject *starred_tuple = PyObject_GC_New(gaobject, &Py_GenericAliasType);
+    starred_tuple->origin = alias->origin;
+    starred_tuple->args = alias->args;
+    starred_tuple->parameters = alias->parameters;
+    starred_tuple->weakreflist = alias->weakreflist;
+    starred_tuple->starred = 1;
+    Py_INCREF(alias->args);
+    // We may not have any parameters to deal with - e.g. `*tuple[int]`
+    if (alias->parameters != NULL) {
+        Py_INCREF(alias->parameters);
+    }
+    _PyObject_GC_TRACK(starred_tuple);
+    return (PyObject *) starred_tuple;
+}
+
+static void
+ga_iter_dealloc(gaiterobject *gi) {
+    _PyObject_GC_UNTRACK(gi);
+    Py_XDECREF(gi->obj);
+    PyObject_GC_Del(gi);
+}
+
+PyTypeObject Py_GenericAliasIterType = {
```

Comment:
My C is rusty, but IIRC, this will make `Py_GenericAliasIterType` public to the C API, and we probably don't want that for now until things stabilize a little. Maybe
```suggestion
static PyTypeObject Py_GenericAliasIterType = {
```

Reply:
Done.